README.md

```diff
@@ -1,61 +1,46 @@
 # SecretHub Cross Language Golang Client<sup>**experimental**</sup>
 
-`secrethub-xgo` wraps the `secrethub-go` client with `cgo` exported functions so it can be called from other languages, e.g. C, Python, Ruby, NodeJS, and Java. 
+`secrethub-xgo` wraps the `secrethub-go` client with `cgo` exported functions so it can be called from other languages, e.g. C, C#, Python, Ruby, NodeJS, and Java. 
 
-For a proof of concept of how a client writting in another language can use `secrethub-xgo`, see the `clients/` folder. Right now it only contains a Java example, but more will be added in the future.
+At the moment, we provide the library for C# usage. 
 
-Make sure you build the `secrethub-xgo` client first before using the example client code. 
+In order to make use of the package, you will need to have the following installed:
+ - [.NET Core](https://docs.microsoft.com/en-gb/dotnet/core/install/)
+ - [Golang](https://golang.org/doc/install)
 
-To build the `secrethub-xgo` c-shared object, run the `build.sh` script. It will place the build in a `linux` or `darwin` folder, depending on your OS. 
+## Create and use the package
 
-You'll need to have [Golang](https://golang.org/doc/install) installed. 
+First, you have to make the NuGet Package and use it in your project. To do so, Here are the steps:
+1. execute `make nupkg` from the Makefile
+2. Go to your .NET project and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
+3. Since our library uses Json.Net, you also need to add this dependency to your project. To do so, run the following command: `dotnet add package Newtonsoft.Json`.
 
+## How to call library functions
 
-### Windows Support
+- Read
+`SecretHub.SecretVersion secret = SecretHub.Client.Read("path_to_secret");`
```

Comment:
This is pretty difficult to read. What about something like the following layout? Then we also directly have a short example in there.

````
### `Read(string path)`
Retrieve a secret, including all its metadata:
```csharp
SecretHub.SecretVersion secret = SecretHub.Client.Read("path/to/secret");
Console.WriteLine("The secret value is " + secret.Data);
```
````

Reply:
And should we add somewhere what is in the `SecretHub.SecretVersion` object?

Reply:
That's actually more readible that what I made before. Thank you for the tip.
Also, I added what `SecretHub.SecretVersion` object is.